client/components/theme-collection/index.tsx

```diff
@@ -0,0 +1,110 @@
+import { Button } from '@wordpress/components';
+import { Icon, chevronLeft, chevronRight } from '@wordpress/icons';
+import { ReactElement, useEffect, useRef, PropsWithChildren, useState } from 'react';
+import './style.scss';
+import { Swiper as SwiperType } from 'swiper/types';
+
+interface ThemeCollectionProps {
+	collectionSlug: string;
+	heading: string;
+	subheading: ReactElement;
+}
+
+export default function ThemeCollection( {
+	collectionSlug,
+	heading,
+	subheading,
+	children,
+}: PropsWithChildren< ThemeCollectionProps > ): ReactElement {
+	const swiperInstance = useRef< SwiperType | null >( null );
+	const swiperContainerId = `swiper-container-${ collectionSlug }`;
+
+	const [ isSwiperLoaded, setSwiperLoaded ] = useState( false );
+	useEffect( () => {
+		if ( ! isSwiperLoaded ) {
```

Comment:
I think this can get this component and swiper out of sync. If the swiper instance gets cleaned up (for whatever reason) then `isSwiperLoaded` will remain true and it won't get recreated.

```suggestion
		if ( ! isSwiperLoaded || !swiperInstance?.current || swiperInstance.current?.destroyed ) {
```